File: Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/m.py
#SVR 

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
Y = dataset.iloc[:, 2:3].values

# Splitting the dataset into the Training set and Test set
"""from sklearn.model_selection import train_test_split
X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size = 0.2, random_state = 0)"""

# Feature Scaling
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
X = sc_X.fit_transform(X)
sc_Y = StandardScaler()
Y = sc_Y.fit_transform(Y)

#Fitting SVR to dataset
from sklearn.svm import SVR
regressor=SVR(kernel='rbf')
regressor.fit(X,Y)

#Predict with SVR
y_pred=regressor.predict(sc_X.transform([[6.5]]))
y_pred=sc_Y.inverse_transform(y_pred)

#Visualizing SVR results
plt.scatter(X,Y,color='red')
plt.plot(X,regressor.predict(X),color='blue')
plt.title('Truth or bluff (SVR model)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

File: Part 2 - Regression/Section 7 - Support Vector Regression (SVR)/m.R
#SVR

#Regression template

#read data
dataset = read.csv('Position_Salaries.csv')
dataset=dataset[,2:3]
dataset$Level2=dataset$Level^2
dataset$Level3=dataset$Level^3
dataset$Level4=dataset$Level^4

#split test and training set
# install.packages('caTools')
# library(caTools)
# set.seed(123)
# split=sample.split(dataset$Purchased,SplitRatio=0.8)
# training_set=subset(dataset,split==TRUE)
# test_set=subset(dataset,split==FALSE)

#feature scaling
#training_set[,2:3]=scale(training_set[,2:3])
#test_set[,2:3]=scale(test_set[,2:3])

#Fitting SVR model to dataset
#install.packages('e1071')
library(e1071)
regressor=svm(formula=Salary~.,data=dataset,type='eps-regression')

#predict data using SVR
y_pred=predict(regressor,data.frame(Level=6.5,
                                    Level2=6.5^2,
                                    Level3=6.5^3,
                                    Level4=6.5^4))


#visualize SVR model
library(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level,y=dataset$Salary),
             colour='red')+
  geom_line(aes(x=dataset$Level,y=predict(regessor,newdata=dataset)),
            colour='blue')+
  ggtitle('True vs bluff (SVR model)')+
  xlab('Level')+
  ylab('Salary')
  























